test/src/test/java/jenkins/model/JenkinsSystemReadAndManagePermissionTest.java

```diff
@@ -0,0 +1,65 @@
+package jenkins.model;
+
+import com.gargoylesoftware.htmlunit.html.HtmlForm;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.MockAuthorizationStrategy;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class JenkinsSystemReadAndManagePermissionTest {
+
+    private static final String SYSTEM_READER = "systemReader";
+
+    @BeforeClass
+    public static void enablePermissions() {
+        System.setProperty("jenkins.security.SystemReadPermission", "true");
+        System.setProperty("jenkins.security.ManagePermission", "true");
+    }
+
+    @AfterClass
+    public static void disablePermissions() {
+        System.clearProperty("jenkins.security.SystemReadPermission");
+        System.clearProperty("jenkins.security.ManagePermission");
+    }
+
+    @ClassRule
+    public static JenkinsRule j = new JenkinsRule();
+
+    private JenkinsRule.WebClient webClient;
+
+    @Before
+    public void setup() {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy()
+                .grant(Jenkins.MANAGE, Jenkins.SYSTEM_READ, Jenkins.READ).everywhere().to(SYSTEM_READER));
+
+        webClient = j.createWebClient();
+        webClient.setThrowExceptionOnFailingStatusCode(false);
+    }
+
+    @Test
+    public void configureReadAllowedWithSystemReadAndManagePermission() throws Exception {
+        HtmlPage configure = webClient.login(SYSTEM_READER)
+                .goTo("configure");
+        assertThat(configure.getWebResponse().getStatusCode(), is(200));
+    }
+
+    @Test
+    public void configureConfigSubmitAllowedWithSystemReadAndManagePermission() throws Exception {
+        HtmlPage configure = webClient.login(SYSTEM_READER)
+                .goTo("configure");
+        assertThat(configure.getWebResponse().getStatusCode(), is(200));
+
+        HtmlForm configureForm = configure.getFormByName("config");
+        HtmlPage submit = j.submit(configureForm);
```

Comment:
Could we get a test that ensures that non-editable fields remain so (e.g. master executors), and that manipulated form submissions aren't effective. Basically, test the `…Unclassified`/`…UnclassifiedReadable` separation.

Probably easiest with a test-only descriptor whose `global.jelly` bypasses the read-only mode, and sets its non-default value directly on the Jelly/Groovy view, so no messing around with the UI in HTMLUnit is needed.

Reply:
done